src/util.js

```diff
@@ -114,8 +115,10 @@ u.boolean = function(s) {
   return s == null || s === '' ? null : s==='false' ? false : !!s;
 };
 
-u.date = function(s) {
-  return s == null || s === '' ? null : Date.parse(s);
+// parse a date with optional format
+u.date = function(s, format) {
+  var d = format ? d3_timeF.format(format) : Date;
```

Comment:
I looked at https://github.com/d3/d3-time-format/blob/master/src/locale.js#L161 and it looked like creating a new formatter is just returning a function. I was curious to see what the impact of returning a function for each value is and it is measurable.

```js
f = require("d3-time-format")
format = f.format("%d.%m.%Y")

var start = new Date().getTime();
for(var i = 0; i < 1000000; i++) {
    format.parse("18.07.1990")
}
var end = new Date().getTime();
var time = end - start;
console.log('Execution time: ' + time);


start = new Date().getTime();
for(var i = 0; i < 1000000; i++) {
    f.format("%d.%m.%Y").parse("18.07.1990")
}
end = new Date().getTime();
time = end - start;
console.log('Execution time: ' + time);
```

```
"Execution time: 7983"
"Execution time: 9602"
```

I rewrote this part as you suggested. The function not takes a d3-time-format format as an optional input.